templates/app_client/lib/app_client/api_version/adapter.rb.erb

```diff
@@ -28,7 +28,10 @@ module <%= gem_name_pascal %>
             config.adapter_factory.connect_oauth(access_token, options)
           elsif api_key && api_secret
             config.adapter_factory.connect(api_secret, options.merge!({
-              user: config.api_key
+              default_headers: {
```

Comment:
These sorts of headers are non-standard, and I think we've made an effort to move to using regular basic auth for these sorts of app-app connections, so I would omit this and continue overriding this in the generated gem if its necessary for the API it's interacting with.